tests/integration/test_smart_contracts.py

```diff
@@ -147,6 +147,30 @@ def test_get_contract_with_contract_id(self):
         self.assertEqual(response.get("status"), 200, response)
         jsonschema.validate(response.get("response"), schema.smart_contract_at_rest_schema)
 
+    def test_get_contract_logs_with_contract_id(self):
+        response = self.client.get_smart_contract_logs(smart_contract_id=SMART_CONTRACT_ARGS_ID)
+        self.assertTrue(response.get("ok"), response)
+        self.assertEqual(response.get("status"), 200)
+        jsonschema.validate(response.get("response"), schema.smart_contract_logs_schema)
+
+    def test_get_contract_logs_with_contract_id_and_tail(self):
+        response = self.client.get_smart_contract_logs(smart_contract_id=SMART_CONTRACT_ARGS_ID, tail=100)
+        self.assertTrue(response.get("ok"), response)
+        self.assertEqual(response.get("status"), 200)
+        jsonschema.validate(response.get("response"), schema.smart_contract_logs_schema)
+
+    def test_get_contract_logs_with_contract_id_and_since(self):
+        response = self.client.get_smart_contract_logs(smart_contract_id=SMART_CONTRACT_ARGS_ID, since="2019-09-16T23:10:04.000Z")
+        self.assertTrue(response.get("ok"), response)
+        self.assertEqual(response.get("status"), 200)
+        jsonschema.validate(response.get("response"), schema.smart_contract_logs_schema)
+
+    def test_get_contract_logs_with_contract_id_tail_and_since(self):
+        response = self.client.get_smart_contract_logs(smart_contract_id=SMART_CONTRACT_ARGS_ID, tail=100, since="2019-09-16T23:10:04.000Z")
```

Comment:
you should test the optional parameters actually changing behavior and assert on it. For example, do a request with tail=1 and assert that your log array is only one in size, and change your since and assert that no logs exist with an older timestamp. The way this is implemented, it tests that adding the parameters returns 200, but doesn't check their functionality